pypuf/simulation.py

```diff
@@ -106,3 +106,36 @@ def ltf_eval(self, inputs):
                 for l in range(self.k)
             ])
         )
+
+
+class NoisyLTFArray(LTFArray):
+    """
+    Class that simulates k LTFs with n bits and a constant term each 
+    with noise effect and constant bias (=scale) added.
+    """
+    def __init__(self, weight_array, transform, combiner, scale, noise):
+        """
+        Initializes LTF array with bias and noise effect based on inputs.
+        The noise effect equals a normal distributed random variable
+        with mu=0, sigma= scale of overall weights * noise.
+        The bias is a constant value out of a normal distribution.
+        """
+        super().__init__(weight_array, transform, combiner)
+        self.scale = scale
+        self.bias = random.normal(loc=0, scale=scale, size=self.k)
```

Comment:
The bias is part of the physical properties of the simulated PUF, which is given to the constructor in the `weight_array`. The purpose to externally generate/determine the `weight_array` and then just store in within the simulation class is that one use case of this class is to give a simulation with pre-defined weights. That's we must support that the user of `NoisyLTFArray` can choose the bias for themselves.

Also, the bias is a feature that should better be added the the (noise-free) parent class, as it is of general interest. I suggest to refactor the `weight_array` to have dimensions `(n+1) by k` and include the bias there. This would be in-line with the mathematical description of Arbiter PUFs in many important papers, including Rührmair et al., Ganji et al.